### DISCUSSION TITLE
How to reduce polyfill costs?

### DISCUSSION THREAD

**Discussion by gergomiklos** [JamesBrill/react-speech-recognition#154] [Q&A]
Hi! First, I would like to tell you that I really like this lib, and thanks!

My question would be about reducing the use of polyfill services, which is Azure's text-to-speech service in my case. Google's free built-in service is almost useless unless you only listen to commands, but other services are extremely expensive ($1+ per hour). I am building a side-project with continuous listening, and accidentally and unfortunately, I spent all my free credits (5 hours) during the development. Is there any method to detect whether the user stopped speaking, and then stop using the polyfill service? Or this is already a built-in behavior, and I missed something? If both answers are no, then can you give me directions on how to implement this "auto-pause" feature?

I am also open to any other idea to reduce the usage of external speech recognition services!

**Comment by johnrizzo4** [comment]
I'm writing a free software for school with speech to text capabilities (to assess pronunciation). I use web speech API through react-speech-recognition. But I need to sw to work off-line too (not all schools have internet). So, I'm implementing a Vosk alternative: 40Mb embedded AI, into my PWA. It would be interesting to implement the needed API layer to have vosk as a provider through polyfill.
Any help on that would be welcome (else, I'll just develop a React hook around vosk, similar to the one of react-speech-recognition).
